engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -986,33 +986,28 @@ public Answer easySend(final Long hostId, final Command cmd) {
     }
 
     @Override
-    public boolean reconnect(final long hostId) {
+    public void reconnect(final long hostId) throws CloudRuntimeException, AgentUnavailableException{
```

Comment:
You added `AgentUnavailableException` to the method signature, but I did not see you throwing this exception anywhere.